src/Components/Components/src/Auth/AuthorizeView.razor

```diff
@@ -0,0 +1,68 @@
+@namespace Microsoft.AspNetCore.Components
+
+@if (currentAuthState == null)
+{
+    @Authorizing
+}
+else if (IsAuthorized())
+{
+    @((Authorized ?? ChildContent)?.Invoke(currentAuthState))
+}
+else
+{
+    @NotAuthorized
+}
+
+@functions {
+    private IAuthenticationState currentAuthState;
+
+    [CascadingParameter] public Task<IAuthenticationState> AuthStateTask { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed if the user is authorized.
+    /// </summary>
+    [Parameter] public RenderFragment<IAuthenticationState> ChildContent { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed if the user is not authorized.
+    /// </summary>
+    [Parameter] public RenderFragment NotAuthorized { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed if the user is authorized.
+    /// If you specify a value for this parameter, do not also specify a value for <see cref="ChildContent"/>.
+    /// </summary>
+    [Parameter] public RenderFragment<IAuthenticationState> Authorized { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed while asynchronous authorization is in progress.
+    /// </summary>
+    [Parameter] public RenderFragment Authorizing { get; private set; }
+
+    protected override async Task OnParametersSetAsync()
+    {
+        // We allow 'ChildContent' for convenience in basic cases, and 'Authorized' for symmetry
+        // with 'NotAuthorized' in other cases. Besides naming, they are equivalent. To avoid
+        // confusion, explicitly prevent the case where both are supplied.
+        if (ChildContent != null && Authorized != null)
```

Comment:
If we want, we can make this a language feature. It would be pretty easy to do if we do it before 3.0.

Reply:
I'm not sure this will be very common, and we could still add it without breaking any existing apps later if it was something components explicitly opted into. Example:

```cs
[Parameter, ParameterAlias("Authorized")] public RenderFragment<IAuthenticationState> ChildContent { get; private set; }
```

So while I think it's a cool idea, I'd prioritise other things.

Reply:
We could add the feature in the future, we wouldn't be able to use it here without a binary breaking change.

Reply:
Ultimately I'm fine with ignoring this for now.